charts/opentelemetry-collector/values.schema.json

```diff
@@ -438,28 +501,94 @@
       ]
     }
   },
-  "anyOf": [
+  "allOf": [
     {
-      "properties": {
-        "standaloneCollector": {
-          "properties": {
-            "enabled": {
-              "const": true
+      "if": {
+        "properties": {
+          "standaloneCollector": {
+            "properties": {
+              "enabled": {
+                "const": true
+              }
+            }
+          }
+        }
+      },
+      "then": {
+        "properties": {
+          "agentCollector": {
+            "properties": {
+              "enabled": {
+                "const": false
+              }
             }
           }
         }
       }
     },
     {
-      "properties": {
-        "agentCollector": {
-          "properties": {
-            "enabled": {
-              "const": true
+      "if": {
+        "properties": {
+          "agentCollector": {
+            "properties": {
+              "enabled": {
+                "const": true
+              }
+            }
+          }
+        }
+      },
+      "then": {
+        "properties": {
+          "standaloneCollector": {
+            "properties": {
+              "enabled": {
+                "const": false
+              }
```

Comment:
Maybe we can use some templates instead of this rule and show meaningful error instead? `{{ fail "" }}` should work I believe. We can put it in NOTES.txt